### PR TITLE
Use the overwrite_from_index_token when pulling the from_index image

### PR DESCRIPTION
**Pull request by mprahl**
This allows IIB to interact with index images it does not have access to by default.

### CODE REVIEW

File: iib/workers/tasks/utils.py [hunk 1]
```diff
@@ -68,6 +73,80 @@ def inner(*args, **kwargs):
...
+            log.debug('The Docker config symlink at %s does not exist', docker_config_path)
+
+        conf = get_worker_config()
+        if os.path.exists(conf.iib_docker_config_template):
+            log.debug(
+                'Copying the Docker config from %s to %s',
+                conf.iib_docker_config_template,
+                docker_config_path,
+            )
+            shutil.copy2(conf.iib_docker_config_template, docker_config_path)
```
Comment (lcarva) [inline iib/workers/tasks/utils.py:131]: Why copy if the file is always re-written a few lines below. Could you just read from `conf.iib_docker_config_template` and write to `docker_config_path` once after modifications?

File: iib/workers/tasks/utils.py [hunk 2]
```diff
@@ -68,6 +73,80 @@ def inner(*args, **kwargs):
...
+    This context manager will reset the authentication to the way it was after it exits. If
+    ``token`` is falsy, this context manager will do nothing.
+
+    :param str token: the token in the format of ``username:password``
+    :param str container_image: the pull specification of the container image to parse to determine
+        the registry this token is for.
+    :return: None
+    :rtype: None
+    """
+    if not token:
```
Comment (lcarva) [inline iib/workers/tasks/utils.py:108]: What happens if request A uses an overwrite token and the clean up fails to execute. When a new request B comes along, which doesn't use an overwrite token, the old authfile will be used, no?

Comment (mprahl) [reply iib/workers/tasks/utils.py:108]: `_cleanup` is always called at the start up of the Celery tasks, which resets the Docker config to the original.

File: iib/workers/tasks/utils.py (hunk 1 above)
Comment (mprahl) [reply iib/workers/tasks/utils.py:131]: Good point. I'll fix that.

**Comment by lcarva** [comment]
Tests need updating due to code change.

**Pull request merged**